.github/actions/configure-environment/action.yml

```diff
@@ -13,11 +13,38 @@ runs:
         /#.*arch=/ && $0 ~ "arch="arch { print }
         ' environment.yml > ci-environment.yml
 
-    - uses: conda-incubator/setup-miniconda@v2
+    - name: Setup Mambaforge
+      uses: conda-incubator/setup-miniconda@v2
       with:
+        miniforge-variant: Mambaforge
+        miniforge-version: latest
+        # Do not specify environment file - see Cache step below
         activate-environment: yardl
-        environment-file: ci-environment.yml
+        use-mamba: true
+
+    - name: Get Date
+      id: get-date
+      shell: bash
+      run: echo "DATE=$(date -u +'%Y%m%d')" >> $GITHUB_ENV
+
+    - name: Cache Conda Env
+      id: cache-conda
+      uses: actions/cache@v3
+      with:
+        path: ${{ env.CONDA }}/envs
+        key:
+            conda-${{ runner.os }}--${{ runner.arch }}--${{
+            hashFiles('ci-environment.yml') }}-${{
+            env.DATE }}-${{
```

Comment:
Why do we want the date to be in the cache key?

Reply:
Adding the date ensure the cache is updated daily to account for any changes in conda package versions. See https://github.com/marketplace/actions/setup-miniconda#caching-environments, which I essentially copied here.